src/stp/new_plays/Attack.cpp

```diff
@@ -0,0 +1,134 @@
+    Dealer::DealerFlag closeToTheirGoalFlag(DealerFlagTitle::CLOSE_TO_THEIR_GOAL, DealerFlagPriority::MEDIUM_PRIORITY);
+    Dealer::DealerFlag notImportant(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::LOW_PRIORITY);
+
+    flagMap.insert({"attacker", {closeToBallFlag}});
+    /*flagMap.insert({"test_role_1", {notImportant}});
+    flagMap.insert({"test_role_2", {notImportant}});
+    flagMap.insert({"test_role_3", {notImportant}});
+    flagMap.insert({"test_role_4", {notImportant}});
+    flagMap.insert({"test_role_5", {notImportant}});
+    flagMap.insert({"test_role_6", {notImportant}});
+    flagMap.insert({"test_role_7", {notImportant}});
+    flagMap.insert({"test_role_8", {notImportant}});
+    flagMap.insert({"test_role_9", {notImportant}});
+    flagMap.insert({"test_role_10", {notImportant}});*/
+
+    auto distribution = dealer.distribute(world->getWorld()->getUs(), flagMap);
+
+    stpInfos = std::unordered_map<std::string, StpInfo>{};
+    for (auto& role : roles) {
+        auto roleName{role->getName()};
+        if (distribution.find(roleName) != distribution.end()) {
+            auto robot = distribution.find(role->getName())->second;
+
+            stpInfos.emplace(roleName, StpInfo{});
+            stpInfos[roleName].setRobot(robot);
+        }
+    }
+}
+
+void Attack::calculateInfoForRoles() noexcept {
+    auto goalTarget = calculateGoalTarget();
+
+    // Calculate attacker info
+    if (stpInfos.find("attacker") != stpInfos.end()) {
+        stpInfos["attacker"].setPositionToShootAt(goalTarget);
+        stpInfos["attacker"].setKickChipType(MAX_SPEED);
+    }
+}
+
+Vector2 Attack::calculateGoalTarget() noexcept {
+    // Position of the ball from which the goal target is determined
+    auto fromPoint = world->getWorld().value().getBall().value()->getPos();
+
+    // Get the longest line section on the visible part of the goal
+    std::vector<Line> openSegments = FieldComputations::getVisiblePartsOfGoal(field, false, fromPoint, world->getWorld().value());
+
+    // If there is no empty location to shoot at, just shoot at the center of the goal
+    if (openSegments.empty()) return field.getTheirGoalCenter();
+
+    // The longest open segment of the goal will be the best to shoot at
+    auto bestSegment = getLongestSegment(openSegments);
+
+    // Make two aim points which are in the corners, since these points are harder for the keeper to intercept
+    Line aimPoints = getAimPoints(field, fromPoint);
+    auto leftPoint = aimPoints.start;
+    auto rightPoint = aimPoints.end;
+
+    // Check if the left and right points are in the best segment
+    double maxY = std::max(bestSegment.start.y, bestSegment.end.y);
+    double minY = std::min(bestSegment.start.y, bestSegment.end.y);
+    bool leftPointInSegment = leftPoint.y < maxY && leftPoint.y > minY;
+    bool rightPointInSegment = rightPoint.y < maxY && rightPoint.y > minY;
```

Comment:
Why is this check necessary? Aren't you choosing it from the segment?

Reply:
We want to shoot at the corners of the goal to make it harder for the keeper, like in real soccer. For that two points on both sides of the goal are calculated. We want to aim for the point that is in the largest segment because that will be the hardest for the keeper to block